openhantek/src/scopesettings.h

```diff
@@ -55,14 +55,19 @@ struct DsoSettingsScopeVoltage {
     double offset = 0.0;        ///< Vertical offset in divs
     double trigger = 0.0;       ///< Trigger level in V
     unsigned gainStepIndex = 6; ///< The vertical resolution in V/div (default = 1.0)
-    union {                     ///< Different enums, coupling for real- and mode for math-channels
-        Dso::MathMode math;
-        unsigned couplingIndex = 0;
-        int rawValue;
-    };
-    QString name;          ///< Name of this channel
-    bool inverted = false; ///< true if the channel is inverted (mirrored on cross-axis)
-    bool used = false;     ///< true if this channel is enabled
+    bool inverted = false;      ///< true if the channel is inverted (mirrored on cross-axis)
+                   ///< Different enums, coupling for real- and mode for math-channels
+    Dso::MathMode math;
+    unsigned couplingIndex = 0;
+    int rawValue;
```

Comment:
math, couplingIndex and rawValue should be in a union. Only rawvalue is read/written by the settings object.

Reply:
The union would not compile on gcc 4.8.5 and all the oscilloscope would not work.

Reply:
With the union and and compiled with gcc 4.8.5 the oscilloscope goes in SIGSEGV once it tries to access any member of the union.

Reply:
Maybe that's a compiler bug. We have a union with an enum class inside. The later is quite new to that compiler version. If you remove the union, math channel and coupling index will never be saved or loaded.